addons/spreadsheet_dashboard/static/src/bundle/dashboard_action/dashboard_loader.js

```diff
@@ -121,24 +147,30 @@ export class DashboardLoader {
                 status: this._getDashboard(dashboard.id).status,
             })),
         }));
+
+        const favoriteDashboards = this.favoriteDashboardIds.map((id) => this.dashboards[id]);
+        if (favoriteDashboards.length) {
+            groups.unshift({
+                id: "favorites",
+                name: "FAVORITES",
+                dashboards: favoriteDashboards,
+            });
+        }
+
+        return groups;
     }
 
     /**
+     * Fetches dashboard groups along with user favorites from the backend.
      * @private
-     * @returns {Promise<{id: number, name: string, published_dashboard_ids: number[]}[]>}
+     * @returns {Promise<{groups: Array<DashboardGroupData>, favorite_dashboard_ids: Array<number>}>}
      */
-    async _fetchGroups() {
-        const groups = await this.orm.webSearchRead(
+    async _fetchDashboardGroupsAndFavorites() {
+        return await this.orm.call(
             "spreadsheet.dashboard.group",
-            [["published_dashboard_ids", "!=", false]],
-            {
-                specification: {
-                    name: {},
-                    published_dashboard_ids: { fields: { name: {} } },
```

Comment:
```
published_dashboard_ids: { fields: { name: {}, is_user_favorite: {} } },
```

Reply:
I did that earlier, but as FRGI mentioned, we also need to preserve the order of the favorited dashboards.

Do you have any suggestions on how we can achieve this using `webSearchRead` while maintaining the order?

Reply:
they should be ordered according to what ? The time they were favorited ? Couldn't find it in the task/spec

Reply:
Yes, The first favorited dashboard should appear at the top of the list, followed by the second, third, and so on.
Had a discussion in task chatter :)

Reply:
An idea would be to add a new computed field `user_favorite_ids` on `spreadsheet.dashboard.group`, add it to the `web_search_read` specification, flatten, then sort the favorites based on their id.
But it's probably a bit far fetched. Let's keep the new `get_dashboard_groups_and_favorites` method